libbeat/metric/system/process/process_linux_common.go

```diff
@@ -0,0 +1,465 @@
+		return "", errors.Wrapf(err, "error fetching user ID for pid %d", pid)
+	}
+	uidValues, ok := status["Uid"]
+	if !ok {
+		return "", fmt.Errorf("Uid not found in proc status")
+	}
+	uidStrings := strings.Fields(uidValues)
+	var userFinal string
+	user, err := user.LookupId(uidStrings[0])
+	if err == nil {
+		userFinal = user.Username
+	} else {
+		userFinal = uidStrings[0]
+	}
+
+	return userFinal, nil
+}
+
+func getEnvData(hostfs resolve.Resolver, pid int, filter func(string) bool) (common.MapStr, error) {
+	path := hostfs.Join("proc", strconv.Itoa(pid), "environ")
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error opening file %s", path)
+	}
+	env := common.MapStr{}
+
+	pairs := bytes.Split(data, []byte{0})
+	for _, kv := range pairs {
+		parts := bytes.SplitN(kv, []byte{'='}, 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		key := string(bytes.TrimSpace(parts[0]))
+		if key == "" {
+			continue
+		}
+
+		if filter == nil || filter(key) {
+
+			env[key] = string(bytes.TrimSpace(parts[1]))
+		}
+
+	}
+	return env, nil
+}
+
+func getMemData(hostfs resolve.Resolver, pid int) (ProcMemInfo, error) {
+	// Memory data
+	state := ProcMemInfo{}
+	path := hostfs.Join("proc", strconv.Itoa(pid), "statm")
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return state, errors.Wrapf(err, "error opening file %s", path)
+	}
+
+	fields := strings.Fields(string(data))
+
+	size, err := strconv.ParseUint(fields[0], 10, 64)
+	if err != nil {
+		return state, errors.Wrapf(err, "error parsing memory size %s", fields[0])
+	}
+	state.Size = opt.UintWith(size << 12)
+
+	rss, err := strconv.ParseUint(fields[1], 10, 64)
+	if err != nil {
+		return state, errors.Wrapf(err, "error parsing memory rss %s", fields[1])
+	}
+	state.Rss.Bytes = opt.UintWith(rss << 12)
+
+	share, err := strconv.ParseUint(fields[2], 10, 64)
+	state.Share = opt.UintWith(share << 12)
+
+	return state, nil
+}
+
+func getCPUTime(hostfs resolve.Resolver, pid int) (ProcCPUInfo, error) {
+	state := ProcCPUInfo{}
+
+	pathCPU := hostfs.Join("proc", strconv.Itoa(pid), "stat")
+	data, err := ioutil.ReadFile(pathCPU)
+	if err != nil {
+		return state, errors.Wrapf(err, "error opening file %s", pathCPU)
+	}
+	fields := strings.Fields(string(data))
+
+	user, err := strconv.ParseUint(fields[13], 10, 64)
+	if err != nil {
+		return state, errors.Wrapf(err, "error parsing user CPU times for pid %d", pid)
+	}
+	sys, err := strconv.ParseUint(fields[14], 10, 64)
+	if err != nil {
+		return state, errors.Wrapf(err, "error parsing system CPU times for pid %d", pid)
+	}
+
+	btime, err := getLinuxBootTime(hostfs)
+	if err != nil {
+		return state, errors.Wrapf(err, "error feting boot time for pid %d", pid)
+	}
+
+	// convert to millis
+	state.User.Ticks = opt.UintWith(user * (1000 / ticks))
+	state.System.Ticks = opt.UintWith(sys * (1000 / ticks))
+	state.Total.Ticks = opt.UintWith(opt.SumOptUint(state.User.Ticks, state.System.Ticks))
+
+	// convert to millis
```

Comment:
This comment is somewhat confusing. When are you converting to millis? If you are doing in both sections why not just have one comment for that?

Reply:
Sorry, that was probably left over from me moving code around.